### ISSUE TITLE
Error when testing connectivity using limit=0 and count=true

### ISSUE THREAD

**Issue by jlanza** [ScorpioBroker/ScorpioBroker#389]
I'm currently testing the [ngsildclient](https://github.com/Orange-OpenSource/python-ngsild-client) which makes a request to the NGSI-LD server to test for connectivity. The request sent is 
```
GET http://scorpio.mydomain.com/ngsi-ld/v1/entities?type=None&limit=0&count=true
```
It reports an error, that seems to be caused by PostSQL.

```
HTTP/1.1 500 Internal Server Error
content-length: 141
connection: close
Content-Type: application/json

{
  "detail": "ERROR: syntax error at or near \"{\" (42601)",
  "type": "https://uri.etsi.org/ngsi-ld/errors/InternalError",
  "title": "Internal error"
}
```

If you modify the URL to include `limit=1` there is no problem.

However it seems that `limit=0&count=true` is supported by the spec and even by Scorpio as if I launch the following request:

```
GET http://scorpio.mydomain.com/ngsi-ld/v1/entities?type=None&limit=0
```
I get the following response, which seems it is considered within the code.

```
HTTP/1.1 400 Bad Request
content-length: 142
connection: close
Content-Type: application/json

{
  "detail": "limit can only be 0 if count is activated",
  "type": "https://uri.etsi.org/ngsi-ld/errors/BadRequestData",
  "title": "Bad Request Data."
}
```

Thanks for the work done.

**Comment by amitrnec** [comment]
Hi,
Now we have fixed this issue #394 

Thanks

**Comment by jlanza** [comment]
Although it seems that right now the error is not arisen, the problem is that the request is not fulfilled correctly.
When launching the request:
```
GET http://scorpio.mydomain.com/ngsi-ld/v1/entities?type=MyType?limit=0&count=true
Accept: application/ld+json
```

I get:
```
HTTP/1.1 200 OK
content-length: 3
connection: close
Content-Type: application/ld+json
count: 0

[]
```

As from the spec I should be getting `NGSILD-Results-Count` header with the actual value, which is not 0. Instead I'm getting a header `count` with value 0, which is not the actual count neither.

**Comment by amitrnec** [comment]
Hi,
Thanks for the update.
you are launching wrong request:
GET http://scorpio.mydomain.com/ngsi-ld/v1/entities?type=MyType?limit=0&count=true
Accept: application/ld+json
it should be 
GET http://scorpio.mydomain.com/ngsi-ld/v1/entities?type=MyType&limit=0&count=true
Accept: application/ld+json

Thanks.

**Comment by jlanza** [comment]
yes you are right. I detected the error, but didn't update the issue. However the problem with the count header label is still there.

I have updated the docker image, but the issue #394 is probably not there. When are you planning to update with the bug fix of #394 . I guess you can easily include the change in the header name ;) I can try to look the file and create a pull request, but you probably can do it much quicker ;)

**Comment by amitrnec** [comment]
Hi,
We have updated the count header label in development-quarkus branch. Docker image will be updated soon.

Thanks.

**Comment by jlanza** [comment]
Thanks for the quick response. I have just updated Scorpio and everything... except from the header tag is working.

**Comment by amitrnec** [comment]
We have updated in code and docker image will be updated soon.

**Comment by ScorpioBroker** [comment]
fixed in 4.0.3
